ddev/ddev
# feat: Add ddev config --update flag, fixes #6035

## The Issue

* #6035 

## How This PR Solves The Issue

Allow overriding after detecting actual project type

## TODO:

- [x] Add a test for --update
- [x] Consider making --update mutually exclusive with other options
- [x] Note to people after `ddev composer create` that they may want to do it.
- [x] drupal8 should not override existing database
- [x] Docs need to explain/expose `ddev config --auto` and `ddev config --update`.  Difference, when to use, etc. How they relate to `ddev config`.
- [ ] Consider changing quickstarts to just use `ddev config --auto` for initial configuration

## Manual Testing Instructions

Assuming you have an existing D9 project, and updating to Drupal 10, you can run 

```console
ddev config --update
```

.ddev/config.yaml changed as expected:

```diff
 name: pantheon-test
-type: drupal9
+type: drupal
 docroot: web
-php_version: "8.1"
+php_version: "8.3"
```

## Automated Testing Overview



## Related Issue Link(s)

Alternative PR to 
* https://github.com/ddev/ddev/pull/6053
* https://github.com/ddev/ddev/pull/5887

## Release/Deployment Notes
